scripts/feature_tests/upgrade/upgrade_vars.sh

```diff
@@ -1,13 +1,19 @@
 #!/bin/bash
+set -eux
 
 # Folder created for specific capi release when running
 # ${CLUSTER_API_REPO}/cmd/clusterctl/hack/create-local-repository.py
 # For CAPI release v0.3.12, the folder v0.3.8 is created
 
 export CAPIRELEASE_HARDCODED="v0.3.8"
 
+function get_capm3_latest() {
+    clusterctl upgrade plan | grep infrastructure-metal3 | awk 'NR == 1 {print $5}'
+}
+
 export CAPM3RELEASE="v0.4.0"
-export CAPM3_REL_TO_VERSION="v0.4.1"
+CAPM3_REL_TO_VERSION="$(get_capm3_latest)" || true
```

Comment:
What is the significance of `|| true` here? Shouldn't we fail the test if we cannot retrieve CAPM3 version?

Reply:
This script is sourced twice. [First time](https://github.com/metal3-io/metal3-dev-env/blob/master/scripts/feature_tests/setup_env.sh#L9), when there is no clusterctl installed yet to do `clusterctl upgrade plan` so it will fail but we just ignore and move on because, we do not need this var at this point but need other vars from the file to be set in the beginning. Then it is [sourced](https://github.com/metal3-io/metal3-dev-env/blob/master/scripts/feature_tests/upgrade/upgrade.sh#L8) once again when actual upgrade task starts and at this point we can normally source it and assign the value to var.